hawc/apps/animal/managers.py

```diff
@@ -297,6 +298,105 @@ def migrate_terms(self, assessment: Assessment) -> pd.DataFrame:
+            invalid_endpoint_ids_str = ", ".join(str(_) for _ in invalid_endpoint_ids)
+            raise ValidationError(f"Invalid id(s) {invalid_endpoint_ids_str}")
+        # endpoints must be in the same assessment
+        assessment = endpoints.first().assessment
+        invalid_endpoints = endpoints.exclude(assessment_id=assessment.id)
+        if invalid_endpoints.exists():
+            invalid_endpoints_str = ", ".join(str(_.pk) for _ in invalid_endpoints)
+            raise ValidationError(
+                f"Endpoints must be from the same assessment; endpoint(s) {invalid_endpoints_str} not from assessment {assessment.id}"
+            )
+        # assessment must have vocab
+        if assessment.vocabulary is None:
+            raise ValidationError("Vocabulary not set in assessment {assessment.id}")
+        # restructure dict for easier seperation of id and terms
+        endpoint_id_to_terms = {obj.pop("id"): obj for obj in objs}
+        # obj fields must be terms
+        term_fields = {
+            "system_term_id",
+            "organ_term_id",
+            "effect_term_id",
+            "effect_subtype_term_id",
+            "name_term_id",
+        }
+        for term_field_to_id in endpoint_id_to_terms.values():
+            if not set(term_field_to_id.keys()).issubset(term_fields):
+                invalid_fields_str = ", ".join(set(term_field_to_id.keys()) - term_fields)
+                raise ValidationError(f"Invalid field name(s) '{invalid_fields_str}'")
+        # all terms must be in assessment vocabulary
+        term_fields_and_ids = {
+            term_field_and_id
+            for term_field_to_id in endpoint_id_to_terms.values()
+            for term_field_and_id in term_field_to_id.items()
+        }
+        term_ids = [id for (_, id) in term_fields_and_ids]
+        terms = Term.objects.filter(pk__in=term_ids)
+        for term in terms:
+            if term.namespace != assessment.vocabulary:
+                raise ValidationError(
+                    f"Assessment vocabulary {assessment.vocabulary} does not match term {term.id} namespace {term.namespace}"
+                )
+        # all terms must match assigned type
+        term_types = VocabularyTermType.as_dict()
+        term_type_to_field = {
+            "system": "system_term_id",
+            "organ": "organ_term_id",
+            "effect": "effect_term_id",
+            "effect_subtype": "effect_subtype_term_id",
+            "endpoint_name": "name_term_id",
+        }
+        term_field_to_text_field = {
+            "system_term_id": "system",
+            "organ_term_id": "organ",
+            "effect_term_id": "effect",
+            "effect_subtype_term_id": "effect_subtype",
+            "name_term_id": "name",
+        }
+        actual_term_fields_and_ids = {
+            (term_type_to_field[term_types[term.type]], term.pk) for term in terms
+        }
+        for term_field_and_id in term_fields_and_ids:
```

Comment:
so i read over the spec and I think I wasn't clear here, my apologies.

A user can only provide an `endpoint_id` and a `name_term_id` for this api endpoint, and given that valid `name_term_id`, it will automatically change all other fields (system/organ/effect/effect-subtype/name) based on the parent of the name-term specified. 

It might be most efficient to just grab the `Term.ehv_dataframe` and then do queries off that, once you've established all the   name_term_id fields are valid.

Apologies with the imprecision here w/ my spec.